code/controller/astar.h

```diff
@@ -0,0 +1,40 @@
+#ifndef ASTAR_H
+#define ASTAR_H
+
+#include <vector>
+#include <map>
+#include <stack>
+
+class Astar {
+public:
+    struct Coord {
+        int x;
+        int y;
+    };;
+
+    typedef typename std::vector< std::vector<int> > grid;
+    typedef typename std::pair<double, Coord> associatedCost;
+
+private:
+    std::vector<Coord> path;
+
+    double manhattanDist(Coord, Coord);
+    bool isValid(int, int, int, int);
+
+    std::vector<Coord> getNeighbours(Coord, int, int);
+    grid makeTerrain();
+
+    void backtrack(Coord, Coord, std::map<Coord, Coord>);
+
+public:
+    Astar();
+    void searchPath(grid terrain, Coord cur, Coord dest);
```

Comment:
how do you get the path after it's calculateddddddddddddddddddddddddddddddddd

Reply:
backtrack adds it to the vector path (I'll change it to a stack)